src/clock.rs

```diff
@@ -0,0 +1,11 @@
+//! A configurable source of time.
+//!
+//! This module provides an API to get the current instant in such a way that
+//! the source of time may be configured. This allows mocking out the source of
+//! time in tests.
+//!
+//! See [Module `tokio_timer::clock`] for full documentation.
+//!
+//! [Module `tokio_timer::clock`]: https://docs.rs/tokio-timer/latest/tokio_timer/clock/index.html
+
+pub use tokio_timer::clock::*;
```

Comment:
As this is going to become part of our API, I suggest we re-export concrete things instead.
I doubt we're going to switch away from tokio, but who knows...